### PR TITLE
Add documentation for enabling Akri to connect to k3s and MicroK8s CNI

### PR DESCRIPTION
**Pull request by bfjelds**
Akri uses the CNI socket and crictl on the host to query some container information.

* Add connection information for K3s' embedded crictl implementation.
* Add connection information for MicroK8s (including crictl installation details)

### CODE REVIEW

**Review by Britel — APPROVED** [review]

File: docs/end-to-end-demo.md [hunk 1]
```diff
@@ -45,6 +45,16 @@ This demo will demonstrate end to end Akri flow, all the way from discovering lo
     ```sh
     kubectl label node ${HOSTNAME,,} node-role.kubernetes.io/master= --overwrite=true
     ```
+1. Akri depends on crictl to track some Pod information and becase Microk8s does not install crictl locally, crictl must be installed and the Akri Helm chart needs to be configured with the crictl path and Microk8s containerd socket
```
Comment (kate-goldenring) [inline docs/end-to-end-demo.md:48]: "because", "MicroK8s"

Comment (kate-goldenring) [inline docs/end-to-end-demo.md:48]: MicroK8s

Comment (kate-goldenring) [inline docs/end-to-end-demo.md:48]: Bit of a run on. How about: "Akri depends on crictl to track some Pod information. MicroK8s does not install crictl locally, so crictl must be installed and the Akri Helm chart needs to be configured with the crictl path and the Microk8s containerd socket.

File: docs/end-to-end-demo.md [hunk 2]
```diff
@@ -75,6 +85,10 @@ This demo will demonstrate end to end Akri flow, all the way from discovering lo
     ```sh
     kubectl label node ${HOSTNAME,,} node-role.kubernetes.io/master= --overwrite=true
     ```
+1. K3s uses its own embedded crictl, so we need to configure the Akri Helm chart with the k3s crictl path and socket
```
Comment (kate-goldenring) [inline docs/end-to-end-demo.md:88]: nit: missing period

File: docs/user-guide.md [hunk 3]
```diff
@@ -29,15 +29,24 @@ kubectl label node $HOSTNAME node-role.kubernetes.io/master= --overwrite=true
...
+    # Note that we aren't aware of any version restrictions
+    VERSION="v1.17.0"
+    curl -L https://github.com/kubernetes-sigs/cri-tools/releases/download/$VERSION/crictl-${VERSION}-linux-amd64.tar.gz --output crictl-${VERSION}-linux-amd64.tar.gz
+    sudo tar zxvf crictl-$VERSION-linux-amd64.tar.gz -C /usr/local/bin
+    rm -f crictl-$VERSION-linux-amd64.tar.gz
+
+    export AKRI_HELM_CRICTL_CONFIGURATION="--set agent.host.crictl=/usr/local/bin/crictl --set agent.host.dockerShimSock=/var/snap/microk8s/common/run/containerd.sock"
     ```
-1. If using K3s, point to `kubeconfig` for Helm.
+1. If using K3s, point to `kubeconfig` for Helm and configure Akri to use the K3s embedded crictl
```
Comment (kate-goldenring) [inline docs/user-guide.md:46]: This makes it seem like the only options are MicroK8s or K3s. Should we add another bullet after that says "If using native K8s, crictl is already installed and no further steps are needed"

**Review by kate-goldenring — CHANGES_REQUESTED** [review]

**Comment by kate-goldenring** [comment]
Maybe not for this PR but we may want to add crictl to our CI workflows since they use MicroK8s

File: docs/user-guide.md (hunk 3 above)
Comment (bfjelds) [reply docs/user-guide.md:46]: will do ... i think this needs revisiting generally.  we should have cluster setup pages for each runtime.

**Review by kate-goldenring — APPROVED** [review]

**Pull request merged**